SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -176,7 +176,27 @@ void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject&
 		// Scoped to exclude OnRep callbacks which are already tracked per OnRep function
 		SCOPE_CYCLE_COUNTER(STAT_ReaderApplyPropertyUpdates);
 
-		for (uint32 FieldId : UpdatedIds)
+		// If we are applying initial data, they must have come from a ComponentData (as it currently stands).
+		// ComponentData will be missing fields if they are completely empty (options, lists, and maps).
+		// However, we still want to apply this empty data, so we need to reconstruct the full
+		// list of field IDs for that component type (Data, OwnerOnly).
+		const TArray<Schema_FieldId>& IdsToIterate = bIsInitialData ?
+			ClassInfoManager->GetFieldIdsByComponentId(ComponentId, *Replicator->RepLayout.Get()) : UpdatedIds;
+#if DO_GUARD_SLOW
```

Comment:
Why are we making this check?

Reply:
I guess this just to be super-safe, maybe in case our assumptions change in the future so that we maybe catch it.
It is quite expensive, which is why it's wrapped in the slow check.

Reply:
True it only runs in debug builds, but I suspect this is still over cautious, or at least a symptom that we aren't encapsulating something well. If we are worried about the implementation of finding all the fields IDs from a schema object that should probably be solved by a test rather than a quadratic invariant check.

Reply:
I think we can back the assumption that a component data fields will be a subset of the total set of fields on the data object.

The place where this would break is if we start adding our own metadata to components, but then it would be correct that the set fields wouldn't be the same as what this function wants to read.

Reply:
The reason the check was added was because we're relying on Unreal's RepState logic to infer schema data fields. We're already making assumptions about how we map the two. This validation is a nice assurance and if we find no issue with this code we could potentially remove it in the long term.